addons/account_facturx/models/account_move.py

```diff
@@ -239,6 +239,21 @@ def find_partner(partner_type):
                                 invoice_line_form.price_unit = float(line_elements[0].text) / float(quantity_elements[0].text)
                             else:
                                 invoice_line_form.price_unit = float(line_elements[0].text)
+                            # For Gross price, we need to check if a discount must be taken into account
+                            line_elements = element.xpath('.//ram:AppliedTradeAllowanceCharge/ram:CalculationPercent',
+                                                          namespaces=tree.nsmap)
+                            if line_elements:
+                                invoice_line_form.discount = float(line_elements[0].text)
+                            else:
```

Comment:
Just one thing. Here you will always try to calculate the discount if you don't find the percentage, I would add a check to see if the `ram:AppliedTradeAllowanceCharge` exists first as you don't want to try to calculate and set values for the discount if no discount exists in the XML.